Documentation/Contents/Pulse.md

```diff
@@ -0,0 +1,286 @@
+
+"shall we look at the code?"
+
+```swift
+  var messagePulse: Pulse<String?> = Pulse(wrappedValue: "Hello tokijh")
+
+  let oldMessagePulse: Pulse<String?> = message
+  message = "Hello tokijh"
+
+  oldMessagePulse != messagePulse // true
+  oldMessagePulse.value == messagePulse.value // true
+```
+
+Well... what is it?
+This looks similar to [distinctUntilChanged](https://reactivex.io/documentation/operators/distinct.html) operator in RxSwift in my think.
+
+and I took the code and ran it in xcode.
+
+<img width="1088" alt="image" src="https://user-images.githubusercontent.com/85085822/162379923-b3ae5b31-be7c-4bda-8ed8-91b643448ea6.png">
+
+Well, there's an error...( An error-free modified code is at the end.)
+
+If so, we'll have no choice but to look at the following documents:
+
+```swift
+  // Reactor
+  private final class MyReactor: Reactor {
+    struct State {
+      @Pulse var alertMessage: String?
+    }
+
+    func mutate(action: Action) -> Observable<Mutation> {
+      switch action {
+      case let .alert(message):
+        return Observable.just(Mutation.setAlertMessage(message))
+      }
+    }
+
+    func reduce(state: State, mutation: Mutation) -> State {
+      var newState = state
+
+      switch mutation {
+      case let .setAlertMessage(alertMessage):
+        newState.alertMessage = alertMessage
+      }
+
+      return newState
+    }
+  }
+
+  // View
+  reactor.pulse(\.$alertMessage)
+    .compactMap { $0 } // filter nil
+    .subscribe(onNext: { [weak self] (message: String) in
+      self?.showAlert(message)
+    })
+    .disposed(by: disposeBag)
+
+  // Cases
+  reactor.action.onNext(.alert("Hello"))  // showAlert() is called with `Hello`
+  reactor.action.onNext(.alert("Hello"))  // showAlert() is called with `Hello`
+  reactor.action.onNext(.doSomeAction)    // showAlert() is not called
+  reactor.action.onNext(.alert("Hello"))  // showAlert() is called with `Hello`
+  reactor.action.onNext(.alert("tokijh")) // showAlert() is called with `tokijh`
+  reactor.action.onNext(.doSomeAction)    // showAlert() is not called
+```
+
+Looking at the `// Cases`, perhaps something similar to '[distinctUntilChanged](https://reactivex.io/documentation/operators/distinct.html)' is correct.
+
+
+```swift
+//
+//  Pulse.swift
+//  ReactorKit
+//
+//  Created by tokijh on 2021/01/11.
+//
+
+@propertyWrapper
+public struct Pulse<Value> {
+
+  public var value: Value {
+    didSet {
+      self.riseValueUpdatedCount()
+    }
+  }
+  public internal(set) var valueUpdatedCount = UInt.min
+
+  public init(wrappedValue: Value) {
+    self.value = wrappedValue
+  }
+
+  public var wrappedValue: Value {
+    get { return self.value }
+    set { self.value = newValue }
+  }
+
+  public var projectedValue: Pulse<Value> {
+    return self
+  }
+
+  private mutating func riseValueUpdatedCount() {
+    if self.valueUpdatedCount == UInt.max {
+      self.valueUpdatedCount = UInt.min
+    } else {
+      self.valueUpdatedCount += 1
+    }
+  }
```

Comment:
This codes updated in https://github.com/ReactorKit/ReactorKit/pull/199.
How about explain with `&+=`?